recipes/abseil/all/conanfile.py

```diff
@@ -175,9 +183,35 @@ def package_info(self):
             self.cpp_info.components[pkgconfig_name].requires = values.get("requires", [])
             if self._is_msvc and self.settings.compiler.get_safe("cppstd") == "20":
                 self.cpp_info.components[pkgconfig_name].defines.extend(["_HAS_DEPRECATED_RESULT_OF", "_SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING"])
-
+            
             self.cpp_info.components[pkgconfig_name].names["cmake_find_package"] = cmake_target
             self.cpp_info.components[pkgconfig_name].names["cmake_find_package_multi"] = cmake_target
 
         self.cpp_info.names["cmake_find_package"] = "absl"
         self.cpp_info.names["cmake_find_package_multi"] = "absl"
+
+        cxx_std_build_module = os.path.join(self.package_folder, self._module_path, self._cxx_std_build_module)
+        self.cpp_info.set_property("cmake_build_modules", cxx_std_build_module)
+        self.cpp_info.components["absl_config"].build_modules["cmake_find_package"] = [cxx_std_build_module]
+        self.cpp_info.components["absl_config"].build_modules["cmake_find_package_multi"] = [cxx_std_build_module]
```

Comment:
Populate some `user_info` with the value of cxx_std? Can it be consumed downstream in some `validate()`?

Reply:
The CMakeDeps generator expects "cmake_build_modules" to be a list of folders, so for me this results in an assertion because join_paths now iterates over the string characters instead of a list:

```
  File "/usr/local/lib/python3.9/site-packages/conan/tools/cmake/cmakedeps/templates/target_data.py", line 277, in __init__
    self.build_modules_paths = join_paths(build_modules)
  File "/usr/local/lib/python3.9/site-packages/conan/tools/cmake/cmakedeps/templates/target_data.py", line 212, in join_paths
    assert os.path.isabs(p), "{} is not absolute".format(p)

```

Reply:
So shouldn't this be `self.cpp_info.set_property("cmake_build_modules", [ cxx_std_build_module ])`?

Reply:
Workaround (without patching the recipe locally):

```
--- tools/cmake/cmakedeps/templates/target_data.py.orig	2022-05-01 12:18:18.000000000 +0200
+++ tools/cmake/cmakedeps/templates/target_data.py	2022-05-01 12:17:04.000000000 +0200
@@ -207,6 +207,7 @@
             e.g: set(LIBFOO_INCLUDE_DIRS "/path/to/included/dir" "/path/to/included/dir2")
             """
             ret = []
+            paths = [paths] if isinstance(paths, str) else paths
             for p in paths:
                 assert os.path.isabs(p), "{} is not absolute".format(p)

```

Reply:
Let´s create a PR to fix it on the recipe side (feel free to post PR for Conan client as well).

Reply:
IMHO, If conan expects this to be a list, it should be list, so fixing this of the recipe side should be sufficient.

Reply:
indeed, it's not a conan client issue, feel free to submit a PR here to fix abseil recipe.